cEP-0021.md

```diff
@@ -0,0 +1,161 @@
+# Profile Bears
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 0021                                              |
+| Version  | 1.0                                               |
+| Title    | Profile Bears                                     |
+| Authors  | Vaibhav Kumar Rai <mailto:raivaibhav08@gmail.com> |
+| Status   | Proposed                                          |
+| Type     | Process                                           |
+
+## Abstract
+
+This cEP describes the detailed implementation of a profiling interface as a
+part of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#6109762077327360).
+
+## Introduction
+
+Profiling a Python program is a form of dynamic analysis that measures the
+execution time of the program and the functions it is composed of. Its goal is
+to produce data about where our program is spending time, and what area might
+be worth optimizing. Thus we can provide a *Profiling interface* from within
+coala and make it available to bear writers so that they can create better
+performant bears.
+
+This cEP proposes to introduce the profiling interface so that bear writers
+will have the ability to Profile the Bear's code to optimize its
+performance.
+
+## Proposed Change
+
+Bear writers can invoke the profiler with the `--profile-bears` argument.
+
+Bear writer will have the ability to directly dump the raw
+profile output (binary output) either on *current working directory* or to a
+*specified directory name*, which can be further used for examination
+of profiler stats with the help of different modules like
+`pstats` or `snakeviz`.
+
+- `--profile-bears` or `profile_bears` (using `.coafile`) is the main
+  argument to enable profiling. It accepts an additional parameter
+  `Dirname` through which bear writers can specify where to store the dump.
+
+The profiler will start by profiling the `run()` method of the bears
+The advantage of profiling just the `run()` method is that bear writer
+don't have to spend time on loading the files, collecting the settings,
+etc. as this is all done by coala itself. They just have to profile
```

Comment:
I am not sure about this but are these bears, files or settings specific? If yes, then instead of using `the` before bears in line 45 and files and settings in line 47, use specific names otherwise remove `the`.